usr/lib/common/obj_mgr.c

```diff
@@ -1696,6 +1696,89 @@ CK_RV object_mgr_restore_obj_withSize(STDLL_TokData_t *tokdata, CK_BYTE *data,
     return rc;
 }
 
+/**
+ * Save the token object to disk and update the shared memory segment.
+ */
+CK_RV object_mgr_save_token_object(STDLL_TokData_t *tokdata, OBJECT *obj)
+{
+    TOK_OBJ_ENTRY *entry = NULL;
+    CK_ULONG index;
+    CK_RV rc;
+
+    // I still think there's a race condition here if two processes are
+    // updating the same token object at the same time.  I don't know how
+    // to solve this short of assigning each token object it's own mutex...
```

Comment:
I guess these comments can be removed. This race should no longer exist (we have object level locking now).

Reply:
Done.